lib/policies.py

```diff
@@ -147,27 +153,69 @@ def printHostToConnMap(self):
 
 class TopologyAwareLoadBalancer(ClusterAwareLoadBalancer):
 
+    PRIMARY_PLACEMENTS = 1
+    FIRST_FALLBACK = 2
+    REST_OF_CLUSTER = -1
+    MAX_PREFERENCE_VALUE = 10
+
     def __init__(self, placementvalues):
+        self.hostToNumConnMap = {}
+        self.hostPortMap = {}
+        self.hostPortMap_public = {}
         self.placements = placementvalues
-        self.allowedPlacements = []
-        self.populatePlacementMap()
-    
+        self.allowedPlacements = {}
+        self.fallbackPrivateIPs = {}
+        self.fallbackPublicIPs = {}
+        self.parseGeoLocations()
+
+    def parseGeoLocations(self):
+        values = self.placements.split(',')
+        for value in values:
+            v = value.split(':')
+            if len(v) > 2 or value[-1] == ':':
+               print('Invalid value part of property topology_keys:', value)
```

Comment:
In java, we error out if the format is incorrect. Same should happen here for consistency.